crates/model/src/signature.rs

```diff
@@ -283,6 +307,87 @@ impl<'de> Deserialize<'de> for EcdsaSignature {
     }
 }
 
+/// An intermediate signature reprensentation used for order creation.
+///
+/// This is to account for the fact that for on-chain signatures like pre-sign
+/// and EIP-1271, the address of the signer is required.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[serde(tag = "signingScheme", rename_all = "lowercase")]
+pub enum CreationSignature {
```

Comment:
> Does it make more sense to have the from in the OrderCreation and match the CreationSignature to properly handle each signature type?

The idea of putting it here is that it makes it mandatory for `PreSign` and `Eip1271` signatures, but optional for ECDSA signatures (why `from` is optional for `Eip712` and `EthSign` but mandatory for `PreSign`; also `PreSign` no longer needs any signature bytes, since its "protocol signature encoding" can be derived just from the order owner).